### PR TITLE
Hw 02

### PR DESCRIPTION
**Pull request by andreikasmykou**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,7 +1,51 @@
...
+
 console.log(config.name); // eslint-disable-line no-console
 
 new models.User();
 new models.Product();
+
+const dirWatcher = new DirWatcher();
+const importer = new Importer();
+
+const directoryPath = `${__dirname.slice(0, -4)}src\\data`;
```
Comment (slowpokex) [inline src/app.js]: path.delimiter instead of \\

File: src/modules/dirwatcher.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import EventEmitter from 'events';
+import fs from 'fs';
+
+import { WrongPathError } from '../errors/wrong-path-error';
+
+export class DirWatcher extends EventEmitter{
+  
+  constructor() {
+    super();
+    console.log('DirWatcher module'); // eslint-disable-line no-console
```
Comment (slowpokex) [inline src/modules/dirwatcher.js]: This log still needed?

File: src/modules/dirwatcher.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+   *              When the path is checked for the first time all files treated as new
+   * @this {DirWatcher}
+   * @param {string} path    directory path for watching
+   * @param {number} delay   delay in ms at which method works
+   * @throws {WrongPathError}
+   * @return void
+   */
+  watch(path, delay) {
+    fs.stat(path, (error, stats) => {
+      if (error || !stats.isDirectory()) {
```
Comment (slowpokex) [inline src/modules/dirwatcher.js]: Please separate handling error and !stats.isDirectory expression

File: src/modules/dirwatcher.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        throw new WrongPathError(`Given path ${path} is not a directory`);
+      }
+      setInterval(() => {
```
Comment (slowpokex) [inline src/modules/dirwatcher.js]: Do you clear this interval?

**Review by slowpokex — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 1 above)
Comment (andreikasmykou) [reply src/app.js]: I think you want to say path.sep
Done

File: src/modules/dirwatcher.js (hunk 2 above)
Comment (andreikasmykou) [reply src/modules/dirwatcher.js]: No, I think no.
But but and not redundant for now))
Delete.

File: src/modules/dirwatcher.js (hunk 4 above)
Comment (andreikasmykou) [reply src/modules/dirwatcher.js]: Add method unWach() and clearTimer in app.js

File: src/modules/dirwatcher.js (hunk 3 above)
Comment (andreikasmykou) [reply src/modules/dirwatcher.js]: Done

**Review by slowpokex — APPROVED** [review]

**Pull request merged**